### PR TITLE
Main morgan

### PR DESCRIPTION
**Pull request by morganwynne**
I'm having a blast tackling the error ``spawn_player(): Condition "!is_inside_tree()" is true. Returning: Transform3D()`` but it's going to take more than one day.

### CODE REVIEW

**Comment by morganwynne** [comment]
There's a weird one in here. The set_global_position method throws an error when the scene is not part of the tree:
```
world.gd:31 @ _init_actor(): Condition "!is_inside_tree()" is true. Returning: Transform3D()
  <C++ Source>   scene/3d/node_3d.cpp:308 @ get_global_transform()
```
This is easily fixed by calling ``add_child`` first. This creates a bug where, if the actor is set to spawn at (0, 0), it will spawn on top (inside?) of one of the other actors. We can avoid this currently by not spawning anything at (0, 0).

File: scripts/actor.gd [hunk 1]
```diff
@@ -32,23 +31,25 @@ signal actor_killed(me: Actor)
...
 	movement_direction = (transform.basis * Vector3(input_dir.x, 0, input_dir.y)).normalized()
 
 func _aim_weapon():
-	# side to side rotation
-	rotator.look_at(controller.get_aim_position(), Vector3.UP)
-	
-	#up and down rotation
-	weapon_base.look_at(controller.get_aim_position(), Vector3.UP)
+	var aim_position = controller.get_aim_position()
+	if !aim_position.is_equal_approx(Vector3.UP):
```
Comment (JohnoRobson) [inline scripts/actor.gd:40]: Is that for fixing the `Node origin and target are in the same position, look_at() failed.` error?

File: scripts/world.gd [hunk 2]
```diff
@@ -11,68 +11,74 @@ class_name World
...
 	player_actors.erase(actor)
 	ai_actors.erase(actor)
 
-func get_closest_actor(from_position:Vector3, ignore: Actor = null) -> Actor:
-	var actors = []
-	actors.append_array(player_actors)
-	actors.append_array(ai_actors)
-	
+func get_closest_actor(from_position: Vector3, ignore: Actor = null) -> Actor:
+	var actors = player_actors + ai_actors # Apparently you can concatenate arrays like this - MW 2023-05-15
```
Comment (JohnoRobson) [inline scripts/world.gd:66]: That's good to know!

File: scripts/actor.gd (hunk 1 above)
Comment (morganwynne) [reply scripts/actor.gd:40]: Correct. The look_at function will throw a warning/error if the target vector is approximately up.
```
E 0:00:00:0674   actor.gd:42 @ _aim_weapon(): Up vector and direction between node origin and target are aligned, look_at() failed.
  <C++ Error>    Condition "p_up.cross(p_target - p_pos).is_zero_approx()" is true.
  <C++ Source>   scene/3d/node_3d.cpp:821 @ look_at_from_position()
  <Stack Trace>  actor.gd:42 @ _aim_weapon()
                 actor.gd:76 @ _process()
```

**Pull request merged**